packages/firestore/src/remote/datastore.ts

```diff
@@ -149,6 +150,33 @@ export async function invokeBatchGetDocumentsRpc(
   return result;
 }
 
+export async function invokeRunQueryRpc(
+  datastore: Datastore,
+  query: Query
+): Promise<Document[]> {
+  const datastoreImpl = debugCast(datastore, DatastoreImpl);
+  const { structuredQuery, parent } = datastoreImpl.serializer.toQueryTarget(
+    query.toTarget()
+  );
+  const params = {
+    database: datastoreImpl.serializer.encodedDatabaseId,
+    parent,
+    structuredQuery
+  };
+
+  const response = await datastoreImpl.invokeStreamingRPC<
```

Comment:
Optional: it appears that `response` is actually a container, perhaps rename it to `responses`, `streamingResponse`, etc.?